File: app/src/main/java/at/msd/friehs_bicha/cdcsvparser/db/AppDatabase.java
package at.msd.friehs_bicha.cdcsvparser.db;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.room.Database;
import androidx.room.DatabaseConfiguration;
import androidx.room.InvalidationTracker;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;
import androidx.sqlite.db.SupportSQLiteOpenHelper;

import at.msd.friehs_bicha.cdcsvparser.transactions.Transaction;
import at.msd.friehs_bicha.cdcsvparser.util.Converter;
import at.msd.friehs_bicha.cdcsvparser.wallet.Wallet;
import at.msd.friehs_bicha.cdcsvparser.wallet.WalletTransactionCrossRef;

@Database(entities = {Wallet.class, Transaction.class, WalletTransactionCrossRef.class}, version = 1)
@TypeConverters({Converter.class})
public abstract class AppDatabase extends RoomDatabase {

    private static AppDatabase INSTANCE;

    public static synchronized AppDatabase getInstance(Context context) {
        if (INSTANCE == null) {
            INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                            AppDatabase.class, "my-database-name")
                    .build();
        }
        return INSTANCE;
    }

    public abstract WalletDao walletDao();

    public abstract TransactionDao transactionDao();

    @NonNull
    @Override
    protected SupportSQLiteOpenHelper createOpenHelper(DatabaseConfiguration config) {
        return null;
    }

    @NonNull
    @Override
    protected InvalidationTracker createInvalidationTracker() {
        return null;
    }

    @Override
    public void clearAllTables() {

    }

}

File: app/src/main/java/at/msd/friehs_bicha/cdcsvparser/db/AppDatabase.kt
package at.msd.friehs_bicha.cdcsvparser.db

import android.content.Context
import androidx.room.*
import at.msd.friehs_bicha.cdcsvparser.transactions.CroCardTransaction
import at.msd.friehs_bicha.cdcsvparser.transactions.Transaction
import at.msd.friehs_bicha.cdcsvparser.util.Converter
import at.msd.friehs_bicha.cdcsvparser.wallet.CroCardWallet
import at.msd.friehs_bicha.cdcsvparser.wallet.Wallet

@Database(entities = [Wallet::class, Transaction::class, CroCardWallet::class, CroCardTransaction::class], version = 3)
@TypeConverters(Converter::class)
abstract class AppDatabase : RoomDatabase() {
    abstract fun walletDao(): WalletDao
    abstract fun transactionDao(): TransactionDao

    abstract fun cardWalletDao(): CardWalletDao

    abstract fun cardTransactionDao(): CardTransactionDao


    override fun clearAllTables() {}

    companion object {
        @Volatile
        private var INSTANCE: AppDatabase? = null
        @Synchronized
        fun getInstance(context: Context): AppDatabase? {
            if (INSTANCE == null) {
                INSTANCE = Room.databaseBuilder(context.applicationContext,
                        AppDatabase::class.java, "my-database-name")
                        .fallbackToDestructiveMigration()
                        .build()
            }
            return INSTANCE
        }
    }
}